testsuite/classlibrary/TestServer_clientIDBooted.sc

```diff
@@ -27,6 +27,23 @@ TestServer_clientID_booted : UnitTest {
 		s.sync;
```

Comment:
one question: if you call `s.boot`, is there a way to sneak in an `Synth.new` after `s.serverRunning == true` and before `initTree`? This is the only race condition I can imagine here …

Reply:
Yes, exactly. One could lock sending to the server while it is not notified yet, 
but that would be an if check on every sendMsg. Not sure that is worth it. 
on local servers, that time window is really short, and you would have to have 
sniper timing ;-). 
On local networks, responses I have seen tended to be quick as well. 

Alternatively, nodeIDAllocator could be nil until created in reaction to server response;
then all attempts to start synths before that would fail with an error, which would be intended correct behavior. 
BUT then one could not reserve any nodeIDs before booting ... which could break user code....
But such user code would not work reliably anyway, because we remake the allocator when clientID comes in. 

looks like an item for the boot-sequence rethink list.

Reply:
For the case that maxLogins and clientID remain unchanged, one could also leave the nodeIDallocator as is, i.e. not remake it when server response comes. 
That would make user code as assumed above work, but would be brittle when moving to multi-clients.

Reply:
well, if `serverRunning` is only true after it has sent a `clientID`, then nothing could use it before that.